packages/utils/src/Signers.ts

```diff
@@ -270,13 +270,13 @@ export async function selectSigners<
  * @param selectors One or more selector callbacks, receiving a signer as input and returning `true` in case it meets selection criteria.
  * @returns The first signer for which all selectors returned `true`, or `undefined` if none meet selection criteria.
  */
-export async function selectSigner<
+export function selectSigner<
   SelectedSigner extends AllSigners, // eslint-disable-line no-use-before-define
   AllSigners extends SignerInterface = SignerInterface
 >(
   signers: readonly AllSigners[],
   ...selectors: readonly SignerSelector[]
-): Promise<SelectedSigner | undefined> {
+): SelectedSigner | undefined {
```

Comment:
The change is based on our conversation?

Reply:
I moved this change to https://github.com/KILTprotocol/sdk-js/pull/819